File: src/main/java/frc/utils/Joystick.java
package frc.utils;

import java.util.function.DoubleSupplier;
import edu.wpi.first.math.geometry.Translation2d;

public class Joystick {
    /*
     * radial deadzone, allows moving at slight angles easier
     */
    public static Translation2d deadzone(double deadzone, double x, double y) {
        if (ExtraMath.getMagnitude(x, y) < deadzone) {
            return new Translation2d();
        }
        return new Translation2d(x, y);
    }

    /*
     * class to hold DoubleSupplier for two joysticks
     */
    public static class duelJoystickAxis{
        public DoubleSupplier lx = () -> 0;
        public DoubleSupplier ly = () -> 0;
        public DoubleSupplier rx = () -> 0;
        public DoubleSupplier ry = () -> 0;

        public duelJoystickAxis(DoubleSupplier lx, DoubleSupplier ly, DoubleSupplier rx, DoubleSupplier ry){
            this.lx = lx;
            this.ly = ly;
            this.rx = rx;
            this.ry = ry;
        }
        public duelJoystickAxis() {}
    }
}

File: src/main/kotlin/frc/utils/Joystick.kt
package frc.utils

import edu.wpi.first.math.geometry.Translation2d
import java.util.function.DoubleSupplier

object Joystick {
    @JvmStatic
    fun deadzone(deadzone: Double, x: Double, y: Double): Translation2d {
        if (ExtraMath.getMagnitude(x, y) < deadzone) {
            return Translation2d()
        }
        return Translation2d(x, y)
    }

    class DuelJoystickAxis @JvmOverloads constructor(
        @JvmField var lx: DoubleSupplier = DoubleSupplier { 0.0 },
        @JvmField var ly: DoubleSupplier = DoubleSupplier { 0.0 },
        @JvmField var rx: DoubleSupplier = DoubleSupplier { 0.0 },
        @JvmField var ry: DoubleSupplier = DoubleSupplier { 0.0 }
    )
}
